core/image/compress_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/gapid/core/image"
+	"github.com/google/gapid/core/image/astc"
+)
+
+type testImageInfo struct {
+	fmt *image.Format
+	ext string
+}
+
+func TestCompressors(t *testing.T) {
+	// TODO: We may need S16toU8 converter.
+
+	testImageInfos := []testImageInfo{
+		{astc.RGBA_4x4, ".png"},
+	}
+
+	for _, testImage := range testImageInfos {
+		plainImg, err := getPlainImage(testImage.fmt.Name)
+		if err != nil {
+			t.Error(err)
+			continue
+		}
+
+		compressedImage, err := plainImg.Convert(testImage.fmt)
+		if err != nil {
+			t.Errorf("Failed to convert '%s' from %v to %v: %v", testImage.fmt.Name, plainImg.Format.Name, testImage.fmt.Name, err)
+			continue
+		}
+
+		refImg, err := getRefImage(testImage.fmt.Name)
+		if err != nil {
+			t.Error(err)
+			continue
+		}
+
+		errs := outputDifference(testImage.fmt.Name, refImg, compressedImage)
+		if errs != nil && len(errs) > 0 {
+			for _, err := range errs {
+				t.Error(err)
+			}
+			continue
+		}
+	}
+}
+
+func getRefImage(name string) (*image.Data, error) {
+	imagePath := filepath.Join("test_data", name+".astc")
+
+	imageData, err := ioutil.ReadFile(imagePath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read  '%s': %v", imagePath, err)
+	}
+
+	imageASTC, err := image.ASTCFrom(imageData)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read '%s': %v", name, err)
+	}
+
+	return imageASTC, nil
+}
+
+func getPlainImage(name string) (*image.Data, error) {
```

Comment:
`getInputImage`

Reply:
Actually not, because this function should get a "plain" non compressed image. When it's named input image, it does not clarify that. With more test added, this function will remain as a function only loads a plain, non compressed image where as another function will grow to load images with different formats.

Reply:
plain does not mean uncompressed. For a compression test, input is naturally uncompressed, but if you want it to be `getUncompressedImage` that would be fine, too.